hermes/probe/create/create.go

```diff
@@ -155,27 +157,38 @@ func CreateFile(ctx context.Context, target *probe.Target, fileID int32, fileSiz
 			status = metrics.ProbeFailed
 		}
 		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
-		return fmt.Errorf("CreateFile(id: %d).%q: could not create file %q: %v", fileID, status, fileName, err)
+		return fmt.Errorf("CreateFile(id: %d).%q: could not create file %q: %w", fileID, status, fileName, err)
 	}
 	if err := wc.Close(); err != nil {
 		status = metrics.WriterCloseFailed
 		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
-		return fmt.Errorf("Writer.Close: %v with status %q", err, status)
+		return fmt.Errorf("Writer.Close: %w with status %q", err, status)
 	}
 	status = metrics.Success
 	target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
 
 	// Verify that the file that has just been created is in fact present in the target system
-	query := &storage.Query{Prefix: fileName[:FileNamePrefixSize]}
+	query := &storage.Query{Prefix: fmt.Sprintf(FileNameFormat, fileID, "")}
 	start = time.Now()
 	objIter := client.Bucket(bucketName).Objects(ctx, query)
 	finish := time.Now()
 	obj, err := objIter.Next()
 	if err != nil {
 		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
-		return fmt.Errorf("CreateFile check failed: %v", err)
+		return fmt.Errorf("CreateFile check failed: %w", err)
+	}
+	namesFound := []string{obj.Name}
+	for {
+		if obj == nil {
+			break
+		}
+		namesFound = append(namesFound, obj.Name)
+		obj, err = objIter.Next()
+	}
+	if len(namesFound) > 1 {
```

Comment:
I would add another check: what if it's zero?
Either add another `if` statement or change the condition to `!= 1`.

Reply:
That's caught in obj, err := objIter.Next()
	if err != nil {
		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
		return fmt.Errorf("CreateFile check failed: %w", err)
	}

Reply:
I would simplify it like this:
1. Execute query
2. Get all results with a for loop
3. Check that we got exactly one result

Reply:
and the exit status is FileMissing then which is not equivalent to the error given when too many files are found

Reply:
I changed the order now but I seperated the case no result and too many results to include the status file missing